backend/src/main/java/com/bakdata/conquery/models/config/XodusStoreFactory.java

```diff
@@ -158,157 +166,127 @@ public class XodusStoreFactory implements StoreFactory {
+			);
 
-	@Override
-	public IdentifiableStore<SearchIndex> createSearchIndexStore(String pathName, ObjectMapper objectMapper) {
-		return StoreMappings.identifiable(createStore(findEnvironment(pathName), validator, SEARCH_INDEX, objectMapper));
+			return loadEnvironmentWithMissingStores;
+		});
 	}
 
 	@Override
-	public SingletonStore<PreviewConfig> createPreviewStore(String pathName, ObjectMapper objectMapper) {
-		return StoreMappings.singleton(createStore(findEnvironment(pathName), validator, ENTITY_PREVIEW, objectMapper));
+	public Collection<? extends WorkerStorage> discoverWorkerStorages() {
+		return loadNamespacedStores("worker_", (storePath) -> new WorkerStorageImpl(this, storePath), WORKER_STORES);
 	}
 
 	@Override
-	public Store<String, Integer> createEntity2BucketStore(String pathName, ObjectMapper objectMapper) {
-		return createStore(findEnvironment(pathName), validator, ENTITY_TO_BUCKET, objectMapper);
-	}
-
-	private Environment findEnvironment(File path) {
-		return registry.findOrCreateEnvironment(path, getXodus());
-	}
-
-	private File resolveSubDir(String... subdirs) {
-		Path current = getDirectory();
-
-		for (String dir : subdirs) {
-			current = current.resolve(dir);
-		}
-
-		return current.toFile();
+	public SingletonStore<Dataset> createDatasetStore(String pathName, ObjectMapper objectMapper) {
+		return StoreMappings.singleton(createStore(findEnvironment(pathName), validator, DATASET, objectMapper, metricRegistry));
 	}
 
-	public <KEY, VALUE> Store<KEY, VALUE> createStore(Environment environment, Validator validator, StoreMappings storeId, ObjectMapper objectMapper) {
+	public <KEY, VALUE> Store<KEY, VALUE> createStore(Environment environment, Validator validator, StoreMappings storeId, ObjectMapper objectMapper,
+													  MetricRegistry metricRegistry) {
 		final StoreInfo<KEY, VALUE> storeInfo = storeId.storeInfo();
 		synchronized (openStoresInEnv) {
 
 			if (openStoresInEnv.get(environment).stream().map(XodusStore::getName).anyMatch(name -> storeInfo.getName().equals(name))) {
 				throw new IllegalStateException("Attempted to open an already opened store:" + storeInfo.getName());
 			}
 
-			final XodusStore store = new XodusStore(environment, storeInfo.getName(), this::closeStore, this::removeStore);
+			final XodusStore xodusStoretore = new XodusStore(environment, storeInfo.getName(), this::closeStore, this::removeStore);
+			openStoresInEnv.put(environment, xodusStoretore);
+
+			boolean cacheBinary = cacheBinaryStore.getOrDefault(storeId, Boolean.FALSE);
+
+			CaffeineSpec binaryCacheSpec = CaffeineSpec.parse("");
+			Store<ByteIterable, ByteIterable> binaryStore = cacheBinary ? new CachedStore<>(xodusStoretore, binaryCacheSpec, metricRegistry) : xodusStoretore;
+
+			if (cacheBinary) {
+				// Start caching of binary data in the background
+				Thread loadStoreThread = new Thread(binaryStore::loadData, "Cache binary store %s from %s".formatted(storeId, environment.getLocation()));
```

Comment:
Das verschiebt den loadData task jetzt hinter ein separates Flag an eine mysteriöse Stelle. Das erscheint mir keine gute Idee

Reply:
Aber das ist nur für den BinaryCache, das deserialized loadData ist komplett raus aus dem Startup und in den LoadStorageTask gewandert